FreeRTOS/FreeRTOS
# Update RISC-V QEMU project for GCC

Description
-----------
After GCC version 11.1.0, "zicsr" extension needs to be specified in the compiler flag.
This PR update the makefile to also support older risc-v GCC version.

reference : https://gcc.gnu.org/git/?p=gcc.git;a=commit;h=b03be74bad08c382da47e048007a78fa3fb4ef49

Test Steps
-----------
Use the following risc-v gcc version to build and run without problem
Built with `./configure --with-multilib-generator="rv32imac-ilp32--"`
```
riscv64-unknown-elf-gcc --version
riscv64-unknown-elf-gcc (g2ee5e430018) 12.2.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
SiFive Freedom tools : https://github.com/sifive/freedom-tools/releases
```
riscv64-unknown-elf-gcc --version
riscv64-unknown-elf-gcc (SiFive GCC-Metal 10.2.0-2020.12.8) 10.2.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

```


Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
